variables.tf

```diff
@@ -254,10 +254,22 @@ variable "logs_config" {
   description = "Configuration for the builds to store log data to CloudWatch or S3."
 }
 
+variable "default_permissions_enabled" {
+  type        = bool
+  default     = true
+  description = "When 'true' default base iam permissions to get up and running with codebuild are attached. Those who want a least priveleged policy can instead set to `false` and use the `custom policy` variable."
+}
+
 variable "extra_permissions" {
-  type        = list(any)
+  type        = list(string)
   default     = []
-  description = "List of action strings which will be added to IAM service account permissions."
+  description = "List of action strings which will be added to IAM service account permissions. Only used if `default_permissions_enabled` is set to true."
+}
+
+variable "custom_policy" {
+  type        = string
+  description = "Jsonencoded IAM policy to add to the IAM service account permissions."
+  default     = ""
```

Comment:
For optional strings like this, the Cloud Posse convention is to use a list of string (in this case), with the default being `[]`. We've written this up but the [PR is still open](https://github.com/cloudposse/.github/pull/30/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R133).

Reply:
sure thing, that should play nicely with how this is merged together with the other policies as well.